src/Functions/goodness.jl

```diff
@@ -0,0 +1,325 @@
+end
+
+# NOTE: Functions below are legacy material. They should work properly, however, they are less well vetted than the sensitivity related functions.
+
+function Entropy_Calc(h::Matrix{Int64})
+    #Calcuates entropy based on h matrix given
+    p=transpose(h)./(sum(h))
+    ElementWise_Entropy=(p.*log.(p))
+    temp_p=deepcopy(ElementWise_Entropy)
+    replace!(temp_p, -NaN=>0)
+    H_pwise=-sum(temp_p)
+    #Entropy of Gaussian was not considered
+    ElementWise_Entropy*=-1
+    return (H_pwise,ElementWise_Entropy)
+end   
+
+function Entropy(F::Vector{Vector{Matrix{Float64}}},n_bins::Int64=100,offset::Float64=0.005,make_plot::Bool=false,save_plot::Bool=false,saveplot_name::String="",saveplot_path::String="")
+       #Creates a vector of entropy values, in the case multiple time signatures are given. If not and F is of length one, returns a vector of length one  
+       #Elemetnt-Wise value returned by function retains NaN values and is therefore not able to be differenciated without further tinkering 
+       Entropy_Value_List=[]
+       ElementWise_Entropy_List=[]
+       for t in 1:length(F)
+           #Pre-processing of F vector 
+               F_copy=deepcopy(F[t])
+               for i in 1:length(F_copy)
+                   no_nan=true
+                   for q in 1:3
+                       for w=1:3
+                           if isnan(F_copy[i][q,w])==true
+                               no_nan=false
+                           end
+                       end
+                   end
+                   F_copy[i]=reshape(F_copy[i],(3,3))
+                   F_copy[i]=F_copy[i]/(det(F_copy[i])^(1/3))
+               end
+   
+           #Gets vectors of k1, k2, and k3
+           k1,k2,k3=FToK2AndK3(F_copy)
+           K3_Vector=k3
+           K2_Vector=k2
+   
+           #Generates 2 histogram, and the h matrix which represents the counts in matrix form
+           K2_edges = (0.0-offset, 1.0-offset)
+           K3_edges=(-1-offset, 1.0-offset) 
+           Hist_K2_Edges=K2_edges[1]:((K2_edges[2]-K2_edges[1])/n_bins):K2_edges[2]
+           Hist_K3_Edges=K3_edges[1]:((K3_edges[2]-K3_edges[1])/n_bins):K3_edges[2]
+           Hist_Matrix=fit(Histogram, (map(Float64,K3_Vector), map(Float64,K2_Vector)), (Hist_K2_Edges, Hist_K3_Edges))
+       #Below line defines bins differently, in a way which doesn't quite work but it worth keep around
+           #Hist_Matrix=fit(Histogram,(map(Float64,K2_Vector),map(Float64,K3_Vector)),nbins=(n_bins,n_bins))
+           h=Hist_Matrix.weights
+           #Makes plots, and subsequently saves plots if inputs are set ot do so 
+           if make_plot==true
+       #Below line creates figure slightly differently, using a method that is not the same as what was used to get h
+               # hist = histogram2d(K2_Vector, K3_Vector, nbins=n_bins, color=:viridis, xlims=K2_edges, ylims=K3_edges, background="black")
+               p=(heatmap(h))
```

Comment:
It would be better to separate the plotting to be done outside this function, perhaps in the `test` folder instead of `src`. I don't think adding the heavy Plots package as a dependency for a single function is a good idea when a user can just load Plots externally. Alternatively, if you would like to provide more convenience plotting functionality, we have a Makie extension which we can extend further to support additional plots.